validator_test.go

```diff
@@ -2965,115 +2965,132 @@ type testByteMap map[byte]byte
 type testStringIntMap map[string]int
+type testIntArray [8]int
 
 func TestRequired(t *testing.T) {
 
-	testString := "foobar"
-	testEmptyString := ""
+	//testString := "foobar"
+	//testEmptyString := ""
 	var tests = []struct {
 		param    interface{}
 		expected bool
 	}{
-		{
-			struct {
-				Pointer *string `valid:"required"`
-			}{},
-			false,
-		},
-		{
-			struct {
-				Pointer *string `valid:"required"`
-			}{
-				Pointer: &testEmptyString,
-			},
-			false,
-		},
-		{
-			struct {
-				Pointer *string `valid:"required"`
-			}{
-				Pointer: &testString,
-			},
-			true,
-		},
-		{
-			struct {
-				Addr Address `valid:"required"`
-			}{},
-			false,
-		},
-		{
-			struct {
-				Addr Address `valid:"required"`
-			}{
-				Addr: Address{"", "123"},
-			},
-			true,
-		},
-		{
-			struct {
-				Pointer *Address `valid:"required"`
-			}{},
-			false,
-		},
-		{
-			struct {
-				Pointer *Address `valid:"required"`
-			}{
-				Pointer: &Address{"", "123"},
-			},
-			true,
-		},
-		{
-			struct {
-				TestByteArray testByteArray `valid:"required"`
-			}{},
-			false,
-		},
+		//{
+		//	struct {
+		//		Pointer *string `valid:"required"`
+		//	}{},
+		//	false,
+		//},
+		//{
+		//	struct {
+		//		Pointer *string `valid:"required"`
+		//	}{
+		//		Pointer: &testEmptyString,
+		//	},
+		//	false,
+		//},
+		//{
+		//	struct {
+		//		Pointer *string `valid:"required"`
+		//	}{
+		//		Pointer: &testString,
+		//	},
+		//	true,
+		//},
+		//{
+		//	struct {
+		//		Addr Address `valid:"required"`
+		//	}{},
+		//	false,
+		//},
+		//{
+		//	struct {
+		//		Addr Address `valid:"required"`
+		//	}{
+		//		Addr: Address{"", "123"},
+		//	},
+		//	true,
+		//},
+		//{
+		//	struct {
+		//		Pointer *Address `valid:"required"`
+		//	}{},
+		//	false,
+		//},
+		//{
+		//	struct {
+		//		Pointer *Address `valid:"required"`
+		//	}{
+		//		Pointer: &Address{"", "123"},
+		//	},
+		//	true,
+		//},
+		//{
+		//	struct {
+		//		TestByteArray testByteArray `valid:"required"`
+		//	}{},
+		//	false,
+		//},
+		//{
+		//	struct {
+		//		TestByteArray testByteArray `valid:"required"`
+		//	}{
+		//		testByteArray{},
+		//	},
+		//	false,
+		//},
 		{
 			struct {
 				TestByteArray testByteArray `valid:"required"`
 			}{
-				testByteArray{},
-			},
-			false,
-		},
-		{
-			struct {
-				TestByteArray testByteArray `valid:"required"`
-			}{
-				testByteArray{'1', '2', '3', '4', '5', '6', '7', 'A'},
-			},
-			true,
-		},
-		{
-			struct {
-				TestByteMap testByteMap `valid:"required"`
-			}{},
-			false,
-		},
-		{
-			struct {
-				TestByteSlice testByteSlice `valid:"required"`
-			}{},
-			false,
-		},
-		{
-			struct {
-				TestStringStringMap testStringStringMap `valid:"required"`
-			}{
-				testStringStringMap{"test": "test"},
-			},
-			true,
-		},
-		{
-			struct {
-				TestIntMap testStringIntMap `valid:"required"`
-			}{
-				testStringIntMap{"test": 42},
+				testByteArray{0x00, '2', 0x00, '4', '5', '6', '7', 'A'},
```

Comment:
This is the test case which I'm developing against (hence the commented out code)